### PR TITLE
Get Doxygen documentation to be independent of the build system

### LINKED ISSUE #1509: GN build parity: doxygen
The GN build needs support for building doxygen to reach parity with automake.

### PR DESCRIPTION
**Pull request by vivien-apple**
#### Problem

Right now doxygen is tied to the previous build system. The current PR makes it independent of the build system and users can call `./scripts/helper/doxygen.sh` to generate the documentation in docs/html.

 #### Summary of Changes
 * Add a `./scripts/helper/doxygen.sh`
 * Replace `Doxyfile.in` by a static `Doxyfile` using ENV vars passed by the doxygen.sh script
 * Remove `./scripts/helpers/doxygen-format-check.sh`as it works without it

fixes #1509

### CODE REVIEW

**Review by woody-apple — APPROVED** [review]

**Review by mspang — APPROVED** [review]

File: docs/Doxyfile [hunk 1]
```diff
@@ -2306,16 +2273,16 @@ DIA_PATH               =
 # and usage relations if the target is undocumented or is not a class.
 # The default value is: YES.
 
-HIDE_UNDOC_RELATIONS   = YES
+HIDE_UNDOC_RELATIONS   = FALSE
```
Comment (bzbarsky-apple) [inline docs/Doxyfile:2276]: Why this change?  And why FALSE instead of NO?

Comment (vivien-apple) [reply docs/Doxyfile]: `FALSE` is a mistake obviously.
This change is a leftover of me trying to find the right parameters for Doxygen to generate appropriate call/caller graphs.
I changed it back to `NO`

**Review by BroderickCarlin — APPROVED** [review]
